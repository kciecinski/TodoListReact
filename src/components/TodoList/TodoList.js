import React from 'react';
import PropTypes from 'prop-types';
import { ListGroup } from 'mdbreact';
import TodoItem from '../TodoItem/TodoItem';

function TodoList({ todoItems, onRemoveItem, onSetItemCompleted }) {
  return (
    <ListGroup>
      {todoItems.map(item => (
        <TodoItem
          key={item.id}
          todoItem={item}
          onRemoveItem={onRemoveItem}
          onSetItemCompleted={onSetItemCompleted}
        />
      )) }
    </ListGroup>
  );
}

TodoList.propTypes = {
  todoItems: PropTypes.arrayOf(PropTypes.shape({
    todoTextValue: PropTypes.string,
  })),
  onRemoveItem: PropTypes.func.isRequired,
  onSetItemCompleted: PropTypes.func.isRequired,
};
TodoList.defaultProps = {
  todoItems: [],
};

export default TodoList;
