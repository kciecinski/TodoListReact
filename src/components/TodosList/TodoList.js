import React from 'react';
import PropTypes from 'prop-types';
import TodoItem from '../TodoItem/TodoItem';

function TodoList({ todoItems }) {
  return (
    <>
      {todoItems.map(item => <TodoItem item={item} />) }
    </>
  );
}

TodoList.propTypes = {
  todoItems: PropTypes.arrayOf(PropTypes.shape({
    todoTextValue: PropTypes.string,
  })),
};
TodoList.defaultProps = {
  todoItems: [],
};

export default TodoList;
