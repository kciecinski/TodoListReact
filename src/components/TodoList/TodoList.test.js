import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TodoList from './TodoList';

configure({ adapter: new Adapter() });

describe('<TodoList />', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      todoItems: [{
        completed: false,
        id: 'NJ1HBA5TN',
        text: 'casd',
      }, {
        completed: false,
        id: 'NJ1HBA5TN',
        text: 'casd',
      },
      ],
      removeItem: jest.fn(),
      setTodoDone: jest.fn(),
    };
    wrapper = shallow(<TodoList {...props} />);
  });

  it('renders 2 items', () => {
    expect(wrapper.find('TodoItem')).toHaveLength(2);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
