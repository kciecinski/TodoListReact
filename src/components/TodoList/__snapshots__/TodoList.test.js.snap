// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> matches snapshot 1`] = `
<ListGroup
  tag="ul"
>
  <TodoItem
    key="NJ1HBA5TN"
    onRemoveItem={[MockFunction]}
    onSetItemCompleted={[MockFunction]}
    todoItem={
      Object {
        "completed": false,
        "id": "NJ1HBA5TN",
        "text": "casd",
      }
    }
  />
  <TodoItem
    key="NJ1HBA5TN"
    onRemoveItem={[MockFunction]}
    onSetItemCompleted={[MockFunction]}
    todoItem={
      Object {
        "completed": false,
        "id": "NJ1HBA5TN",
        "text": "casd",
      }
    }
  />
</ListGroup>
`;
